tests/drivers/watchdog/wdt_error_cases/src/main.c

```diff
@@ -0,0 +1,1001 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/ztest_error_hook.h>
+
+/*
+ * To use this test, either the devicetree's /aliases must have a
+ * 'watchdog0' property, or one of the following watchdog compatibles
+ * must have an enabled node.
+ */
+#if DT_NODE_HAS_STATUS(DT_ALIAS(watchdog0), okay)
+#define WDT_NODE DT_ALIAS(watchdog0)
+#elif DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_wdt)
+#define WDT_NODE DT_INST(0, nordic_nrf_wdt)
+#elif DT_HAS_COMPAT_STATUS_OKAY(zephyr_counter_watchdog)
+#define WDT_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(zephyr_counter_watchdog)
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
+#define NOINIT_SECTION ".dtcm_noinit.test_wdt"
+#else
+#define NOINIT_SECTION ".noinit.test_wdt"
+#endif
+
+/* Common for all targets: */
+#define DEFAULT_WINDOW_MAX	 (500U)
+#define DEFAULT_WINDOW_MIN	 (0U)
+#define DEFAULT_OPTIONS		 (WDT_OPT_PAUSE_IN_SLEEP | WDT_OPT_PAUSE_HALTED_BY_DBG)
+#define MAX_INSTALLABLE_TIMEOUTS (8)
+
+/* By default run all error checks,
+ * unset below to align test scope for the specific target
+ */
+#define WDT_FLAG_RESET_NONE_IS_NOT_SUPPORTED
+#define WDT_FLAG_RESET_CPU_CORE_IS_NOT_SUPPORTED
+#define WDT_FLAG_RESET_SOC_IS_NOT_SUPPORTED
+#define WATCHDOG_SUPPORTS_ONLY_ONE_TIMEOUT_VALUE
+#define WDT_OPT_PAUSE_IN_SLEEP_IS_NOT_SUPPORTED
+#define WDT_OPT_PAUSE_HALTED_BY_DBG_IS_NOT_SUPPORTED
+#define WDT_DISABLE_IS_NOT_SUPPORTED
+#define WDT_FEED_CAN_STALL
+
+/* Align tests to the specific target: */
+#if defined(CONFIG_SOC_NRF54L15)
```

Comment:
Does this test really only work with these options on that specific MCU? Otherwise we should select the options based on the driver compatible from Devicetree, e.g.:

```suggestion
#if DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_wdt)
```

Reply:
Set of supported flags/options/features depend on hardware implementation of the watchdog peripheral.
Our driver (nordic_nrf_wdt) can be used with devices that doesn't allow wdt_disable() calls (like nRF51/nRF52).
And, with devices that have such feature (nRF53, nRF54L, nRF54H). Among them are devices that support flag WDT_FLAG_RESET_CPU_CORE, others support WDT_FLAG_RESET_SOC.
I think it will be difficult to deduct supported features from the driver itself.
I add guidelines in Readme.txt to ease enabling these tests on other targets.